src/traits.rs

```diff
@@ -1,4 +1,63 @@
-use crate::{IsoWeek, Weekday};
+use core::ops::{Add, RangeInclusive};
+
+use crate::{Duration, IsoWeek, Weekday};
+
+/// The common set of methods for a week component.
+pub trait Weeklike: Sized {
+    /// An associated type, representing a day within the context of a week.
+    type Day: Add<Duration, Output = Self::Day>;
+
+    /// Returns a date representing the start of the week.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use chrono::{NaiveDate, Weekday, Weeklike};
+    ///
+    /// let date = NaiveDate::from_ymd(2022, 4, 18);
+    /// let week = date.week(Weekday::Mon);
+    /// assert!(week.start_day() <= date);
+    /// ```
+    fn start_day(&self) -> Self::Day;
```

Comment:
How do you feel about renaming `start_day()` and `end_day()` to `first_day()` and `last_day()`? My worry is that `end_day()` could be interpreted as either the last day of the week or the first day of next week, which could be confusing, whereas this is explicit for `last_day()`.